Use proper HostConfig element which contains the proper Memory value …

ecliptik opened the pull request:
…- fixes #1766 

Tested locally and containers deployed properly using _docker_api_version: "1.19"_

Commit:
Use proper HostConfig element which contians the proper Memory value - fixes #1766

bcoca commented:
what happens to those using earlier versions? this should check api version and use one or the other

bcoca commented:
as these show, this is now a conflict between those running diff versions #1744 #1763 #1602 #1766

Commit:
Set the API version when checking differences in containers and use
this to determine the location of the Memory value depending on the version used.

In v1.18 and earlier it was ['Config']['Memory'], but in v1.19 it
changed to ['HostConfig']['Memory'].

ecliptik commented:
Updated the pull request to include an API check and use the proper data structure depending on the API version,
- v1.18 and earlier - ['Config']['Memory']
- v1.19 -  ['HostConfig']['Memory']

This still gives the error which is due to #1707 and looks like it should work in Ansible v1.9.3 or v2.0.

```
  File "/usr/lib/python2.6/site-packages/docker/utils/utils.py", line 546, in create_container_config
    'mem_limit has been moved to host_config in API version 1.19'
docker.errors.InvalidVersion: mem_limit has been moved to host_config in API version 1.19
```

Commit:
Use a if/else instead to avoid loading possibly invalid values for Memory

bcoca commented on cloud/docker/docker.py:
```diff
@@ -1008,7 +1011,11 @@ def get_differing_containers(self):
+            if docker_api_version >= 1.19:
+                actual_mem = container['HostConfig']['Memory']
+            else:
+                actual_mem = container['Config']['Memory']
```
is it not possible to use ensure_capability as we do with other features?
